components/script/dom/globalscope.rs

```diff
@@ -2878,124 +2882,184 @@ impl GlobalScope {
     }
 
     /// Evaluate JS code on this global scope.
-    pub(crate) fn evaluate_js_on_global_with_result(
+    pub(crate) fn evaluate_js_on_global(
         &self,
         code: Cow<'_, str>,
+        filename: &str,
+        introduction_type: Option<&'static CStr>,
         rval: MutableHandleValue,
-        fetch_options: ScriptFetchOptions,
-        script_base_url: ServoUrl,
         can_gc: CanGc,
-        introduction_type: Option<&'static CStr>,
     ) -> Result<(), JavaScriptEvaluationError> {
-        let source_code = SourceCode::Text(Rc::new(DOMString::from_string(code.into_owned())));
-        self.evaluate_script_on_global_with_result(
-            &source_code,
-            "",
-            rval,
-            1,
-            fetch_options,
-            script_base_url,
-            can_gc,
-            introduction_type,
-        )
+        let cx = GlobalScope::get_cx();
+        let ar = enter_realm(self);
+        let _aes = AutoEntryScript::new(self);
+
+        let url = self.api_base_url();
+        let fetch_options = ScriptFetchOptions::default_classic_script(self);
+
+        rooted!(in(*cx) let mut compiled_script = std::ptr::null_mut::<JSScript>());
+        compiled_script.set(compile_script(cx, &code, filename, 1, introduction_type));
+
+        if compiled_script.is_null() {
+            debug!("error compiling Dom string");
+            report_pending_exception(cx, true, InRealm::Entered(&ar), can_gc);
+            return Err(JavaScriptEvaluationError::CompilationFailure);
+        }
+
+        let script = NonNull::new(*compiled_script).expect("Can't be null");
+
+        if !evaluate_script(cx, script, url, fetch_options, rval) {
+            let error_info =
+                take_and_report_pending_exception_for_api(cx, InRealm::Entered(&ar), can_gc);
+            return Err(JavaScriptEvaluationError::EvaluationFailure(error_info));
+        }
+
+        maybe_resume_unwind();
+        Ok(())
     }
 
-    /// Evaluate a JS script on this global scope.
+    /// <https://html.spec.whatwg.org/multipage/#creating-a-classic-script>
     #[expect(unsafe_code)]
-    #[allow(clippy::too_many_arguments)]
-    pub(crate) fn evaluate_script_on_global_with_result(
+    #[expect(clippy::too_many_arguments)]
+    pub(crate) fn create_a_classic_script(
```

Comment:
Please add spec steps, as I think we are missing some (I don't see script creation time here, is that missing for example?)